components/benefit_sponsors/app/policies/benefit_sponsors/person_policy.rb

```diff
@@ -3,10 +3,48 @@
 module BenefitSponsors
   # Policy for person
   class PersonPolicy < ApplicationPolicy
+    ACCESSABLE_ROLES = %w[hbx_staff_role broker_role active_broker_staff_roles].freeze
+
     def can_read_inbox?
-      return true if user.person.hbx_staff_role
-      return true if (user.person&.broker_role || record.broker_role) && (user.person.id == record.id)
+      return true if can_hbx_staff_modify?
+      return true if can_broker_modify?
       false
     end
+
+    def can_broker_modify?
+      (has_broker_role? || has_active_broker_agency_staff_role?) && matches_broker_agency_profile?(record&.broker_role&.benefit_sponsors_broker_agency_profile_id)
+    end
+
+    def matches_broker_agency_profile?(broker_agency_profile_id)
+      Array(role).any? do |role|
+        role.benefit_sponsors_broker_agency_profile_id == broker_agency_profile_id
+      end
+    end
+
+    def can_hbx_staff_modify?
+      role.is_a?(HbxStaffRole)
```

Comment:
@jacobkagon We need to also check if the hbx_staff_role's permission's modify_family is set to true to confirm if the Hbx Staff Role has access to view/modify the family.